Change artifact download to use short checksums

mobileoverlord opened the pull request:
Instead of using the full checksum for the artifact download file name we will use only the first 7 similar to that of the git default. Since the artifact name also contains the version number, uniqueness isn't going to be an issue.

fhunleth commented:
Where did you land on this one and not breaking all current systems that use the entire hash string?

mobileoverlord commented:
I updated the site expander to add both versions for now. It will first try the short sum and then fall through to the long one. I also added some comments to the code regarding cleaning this up sometime following 1.0 since the plan is to only use short sums moving fwd.

Fixes https://github.com/nerves-project/nerves/issues/250

```
Resolving Nerves artifacts...
  Resolving nerves_system_bbb
  => Downloading https://github.com/nerves-project/nerves_system_bbb/releases/download/v0.20.0/nerves_system_bbb-portable-0.20.0-F7599E9.tar.gz
  => Downloading https://github.com/nerves-project/nerves_system_bbb/releases/download/v0.20.0/nerves_system_bbb-portable-0.20.0-F7599E9B2F6A9BCAE5B9E2B98A3440CDFEA6B0FA347F290605C4111E11289C8F.tar.gz
|==================================================| 100% (112 / 112) MB
```
